demos/mask_rcnn_demo/cpp/main.cpp

```diff
@@ -45,240 +49,327 @@ bool ParseAndCheckCommandLine(int argc, char *argv[]) {
+        old_model = true;
+    }
 
-        // -----------------------------------------------------------------------------------------------------
+    std::string boxes_tensor_name;
+    std::string masks_tensor_name;
+    ov::Layout masks_tensor_layout = {"NCHW"};
 
-        // ---------------------------Prepare output blobs------------------------------------------------------
-        InferenceEngine::OutputsDataMap outputInfo(network.getOutputsInfo());
-        for (auto & item : outputInfo) {
-            item.second->setPrecision(InferenceEngine::Precision::FP32);
+    for (ov::Output<ov::Node> output : outputs) {
+        ov::Shape shape = output.get_shape();
+        size_t dims = shape.size();
+        if (dims == 2) {
+            boxes_tensor_name = output.get_any_name();
+        }
+        if (dims == 4) {
+            masks_tensor_name = output.get_any_name();
         }
+    }
 
-        // -----------------------------------------------------------------------------------------------------
+    // Collect images
+    std::vector<cv::Mat> images;
 
-        // -------------------------Load model to the device----------------------------------------------------
-        auto executableNetwork = ie.LoadNetwork(network, FLAGS_d);
-        logExecNetworkInfo(executableNetwork, FLAGS_m, FLAGS_d);
-        slog::info << "\tBatch size is set to " << netBatchSize << slog::endl;
+    if (modelBatchSize > imagePaths.size()) {
+        slog::warn << "Model batch size is greater than number of images (" << imagePaths.size() <<
+            "), some input files will be duplicated" << slog::endl;
+    }
+    else if (modelBatchSize < imagePaths.size()) {
+        modelBatchSize = imagePaths.size();
+        slog::warn << "Model batch size is less than number of images (" << imagePaths.size() <<
+            "), model will be reshaped" << slog::endl;
+    }
 
-        // -------------------------Create Infer Request--------------------------------------------------------
-        auto infer_request = executableNetwork.CreateInferRequest();
+    ov::preprocess::PrePostProcessor ppp(model);
 
-        // -----------------------------------------------------------------------------------------------------
+    if (old_model) {
+        ppp.input(image_tensor_name)
+            .tensor()
+            .set_element_type(ov::element::u8)
+            .set_layout(image_tensor_layout);
+    }
 
-        // -------------------------------Set input data--------------------------------------------------------
-        /** Iterate over all the input blobs **/
-        for (const auto & inputInfoItem : inputInfo) {
-            InferenceEngine::Blob::Ptr input = infer_request.GetBlob(inputInfoItem.first);
+    ppp.input(image_tensor_name)
+        .tensor()
+        .set_layout(image_tensor_layout)
+        .set_spatial_static_shape(
+            image_tensor_height,
+            image_tensor_width);
```

Comment:
```suggestion
        .set_layout(image_tensor_layout);
```
I don't see what this is for

Reply:
I don't see how this helps but ok